packages/rocketchat-lib/server/models/Users.js

```diff
@@ -138,21 +138,14 @@ class ModelUsers extends RocketChat.models._Base {
 		}
 
 		const termRegex = new RegExp(s.escapeRegExp(searchTerm), 'i');
+		const orStmt = _.reduce(_.keys(options.fields), function(acc, el) {
+			acc.push({ [el]: termRegex }); return acc;
+		}, []);
 		const query = {
 			$and: [
 				{
 					active: true,
-					$or: [
-						{
-							username: termRegex
-						},
-						{
-							name: termRegex
-						},
-						{
-							'emails.address': termRegex
```

Comment:
@rodrigok I've made a new commit which I think has a better implementation and fix the issue you commented above.